### PR TITLE
Thread allocationKey into ResolutionDetails flagMetadata

### PR DESCRIPTION
**Pull request by typotter**
### What does this PR do?

The `allocationKey` field from the Precomputed Flags API response (`PrecomputedFlag`) was not included in `ResolutionDetails.flagMetadata`, making it unavailable to callers who use `resolve()` for rich evaluation details (e.g. OpenFeature providers).

This PR threads `allocationKey` into `flagMetadata` so it is accessible alongside other `extraLogging` fields.

### Motivation

OpenFeature providers (e.g. `dd-openfeature-provider-kotlin`) expose evaluation metadata to consumers via the `EvaluationMetadata` map. `allocationKey` is a key field for attribution and debugging. Without it in `flagMetadata`, providers cannot surface it to OpenFeature clients.

### Additional Notes

- `buildMetadata(PrecomputedFlag)` replaces the old `extractMetadata(JSONObject)` helper (dead code removed).
- `allocationKey` is only added when non-blank.
- A new test covers the blank `allocationKey` edge case to verify it is excluded from metadata.

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](../CONTRIBUTING.md) doc)

### CODE REVIEW

**Review by 0xnm — DISMISSED** [review]

**Comment by typotter** [comment]
FFL-2510

File: features/dd-sdk-android-flags-openfeature/src/test/kotlin/com/datadog/android/flags/openfeature/internal/adapters/ConvertersTest.kt [hunk 1]
```diff
@@ -135,6 +135,25 @@ internal class ConvertersTest {
...
+            value = fakeValue,
+            reason = ResolutionReason.TARGETING_MATCH,
+            flagMetadata = mapOf("allocationKey" to fakeAllocationKey)
+        )
+
+        // When
+        val result = resolution.toProviderEvaluation()
+
+        // Then
+        assertThat(result.metadata?.getString("allocationKey")).isEqualTo(fakeAllocationKey)
```
Comment (abrooksv) [inline features/dd-sdk-android-flags-openfeature/src/test/kotlin/com/datadog/android/flags/openfeature/internal/adapters/ConvertersTest.kt]: Fails CI due to metadata is non-null

> e: warnings found and -Werror specified
w: file:///go/src/github.com/DataDog/dd-sdk-android/features/dd-sdk-android-flags-openfeature/src/test/kotlin/com/datadog/android/flags/openfeature/internal/adapters/ConvertersTest.kt:154:35 Unnecessary safe call on a non-null receiver of type EvaluationMetadata

```suggestion
        assertThat(result.metadata.getString("allocationKey")).isEqualTo(fakeAllocationKey)
```

File: features/dd-sdk-android-flags-openfeature/src/main/kotlin/com/datadog/android/flags/openfeature/internal/adapters/Converters.kt [hunk 2]
```diff
@@ -38,19 +40,37 @@ internal fun <T : Any> ResolutionDetails<T>.toProviderEvaluation(): ProviderEval
...
 )
 
+private fun Map<String, Any>.toEvaluationMetadata(): EvaluationMetadata {
+    val builder = Builder()
+    forEach { (key, value) ->
+        when (value) {
+            is String -> builder.putString(key, value)
+            is Boolean -> builder.putBoolean(key, value)
+            is Int -> builder.putInt(key, value)
+            is Double -> builder.putDouble(key, value)
```
Comment (abrooksv) [inline features/dd-sdk-android-flags-openfeature/src/main/kotlin/com/datadog/android/flags/openfeature/internal/adapters/Converters.kt:54]: question: do we not need to worry about Long with builder.putLong? https://github.com/open-feature/kotlin-sdk/blob/cfe7ac44d92aef0366e4e6cb8bd01ce070ea06d7/kotlin-sdk/src/commonMain/kotlin/dev/openfeature/kotlin/sdk/FlagEvaluationMetadata.kt#L48

**Comment by typotter** [comment]
Verified via `javap` on the actual `kotlin-sdk-release-api.jar` for the 0.6.2 version used in this project — `Builder` exposes only `putString`, `putInt`, `putDouble`, `putBoolean`. There is no `putLong`. The link you shared is likely to a different SDK version or branch. The compile error (`Unresolved reference: putLong`) was correct, and the removal is the right fix. Added a test to document that `Long` values fall through to the `else -> putString` branch.

**Review by abrooksv — APPROVED** [review]

File: features/dd-sdk-android-flags-openfeature/src/test/kotlin/com/datadog/android/flags/openfeature/internal/adapters/ConvertersTest.kt [hunk 3]
```diff
@@ -135,6 +135,44 @@ internal class ConvertersTest {
...
+        // Then
+        assertThat(result.metadata.getString("allocationKey")).isEqualTo(fakeAllocationKey)
+    }
+
+    @Test
+    fun `M serialize Long to string W toProviderEvaluation() {flagMetadata contains Long value}`(
+        @BoolForgery fakeValue: Boolean
+    ) {
+        // Given — Builder has no putLong (only putInt/putDouble/putString/putBoolean);
+        // Long values fall through to the else branch and are stored as their toString() representation.
```
Comment (0xnm) [inline features/dd-sdk-android-flags-openfeature/src/test/kotlin/com/datadog/android/flags/openfeature/internal/adapters/ConvertersTest.kt:162]: `long` value can be converted to `int` though if it fits

**Review by 0xnm — APPROVED** [review]
